pkg/operation/etcdencryption/encryptionconfiguration_test.go

```diff
@@ -0,0 +1,269 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package encryptionconfiguration_test
+
+import (
+	. "github.com/gardener/gardener/pkg/operation/etcdencryption"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	apiserverconfigv1 "k8s.io/apiserver/pkg/apis/config/v1"
+)
+
+const (
+	passiveConfig = `
+apiVersion: apiserver.config.k8s.io/v1
+kind: EncryptionConfiguration
+resources:
+- providers:
+  - identity: {}
+  - aescbc:
+      keys:
+      - name: key1557839139300191000
+        secret: kL03D325eDntRBfLtlh0jFSqQE0Ji69bVq421IwJGiI=
+  resources:
+  - secrets`
+
+	activeConfig = `
+apiVersion: apiserver.config.k8s.io/v1
+kind: EncryptionConfiguration
+resources:
+- providers:
+  - aescbc:
+      keys:
+      - name: key1557839139300191000
+        secret: kL03D325eDntRBfLtlh0jFSqQE0Ji69bVq421IwJGiI=
+  - identity: {}
+  resources:
+  - secrets
+`
+
+	activeConfig2 = `
+apiVersion: apiserver.config.k8s.io/v1
+kind: EncryptionConfiguration
+resources:
+- providers:
+  - aescbc:
+      keys:
+      - name: key1557839139300191000
+        secret: kL03D325eDntRBfLtlh0jFSqQE0Ji69bVq421IwJGiI=
+  - identity: {}
+  resources:
+  - secrets
+`
+)
+
+var _ = Describe("encryptionconfiguration", func() {
+
+	Describe("#NewPassiveConfiguration", func() {
+		It("should create a new passive EncryptionConfiguration", func() {
+			_, err := NewPassiveConfiguration()
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	Describe("#Validate", func() {
+		It("should accept a newly created passive EncryptionConfiguration as consistent", func() {
+			ec, err := NewPassiveConfiguration()
+			Expect(err).NotTo(HaveOccurred())
+			err = Validate(ec)
+			Expect(err).NotTo(HaveOccurred())
+		})
+		It("should detect inconsistency (wrong kind)", func() {
+			ec, err := NewPassiveConfiguration()
+			Expect(err).NotTo(HaveOccurred())
+			ec.Kind = "wrong"
+			err = Validate(ec)
+			Expect(err).To(HaveOccurred())
+		})
+		It("should detect inconsistency (wrong APIVersion)", func() {
+			ec, err := NewPassiveConfiguration()
+			Expect(err).NotTo(HaveOccurred())
+			ec.APIVersion = "v1"
+			err = Validate(ec)
+			Expect(err).To(HaveOccurred())
+		})
+		It("should detect inconsistency (no resource configuration)", func() {
+			ec, err := NewPassiveConfiguration()
+			Expect(err).NotTo(HaveOccurred())
+			ec.Resources = nil
+			err = Validate(ec)
+			Expect(err).To(HaveOccurred())
+		})
+		It("should detect inconsistency (too many resource configurations)", func() {
+			ec, err := NewPassiveConfiguration()
+			Expect(err).NotTo(HaveOccurred())
+			ec.Resources = []apiserverconfigv1.ResourceConfiguration{
+				{},
+				{},
+			}
+			err = Validate(ec)
+			Expect(err).To(HaveOccurred())
```

Comment:
For all errors, you should be able to check what exactly occurred (see our validation tests).

Reply:
adapted tests accordingly and am checking the error field